src/Hl7.Fhir.Specification.Tests/Validation/BasicValidationTests.cs

```diff
@@ -1390,6 +1390,49 @@ public void ValidateWithTargetProfileAndChildDefinitions()
 
         }
 
+        [Fact]
+        public async void TestValidationPerformanceUsCore()
+        {
+            var resolver = new MultiResolver(new CachedResolver(new DirectorySource(Path.Combine("TestData", "us-core-profiles"))), new CachedResolver(new ZipSource("specification.zip")));
+
+            var ctx = new ValidationSettings()
+            {
+                ResourceResolver = resolver,
+                GenerateSnapshot = true,
+                EnableXsdValidation = false,
+                Trace = false,
+                ResolveExternalReferences = false
+            };
+
+            _validator = new Validator(ctx);
+
+            var instances = new[] { "us-core-instance-1.json", 
```

Comment:
If you turn this into a theory, with the instances as data, the stopwatchtimings are also correct.
You would however need to have the CachedResolver in a fixture then, or it would keep generating snapshots all the time.